File: web/screenshot.js
import puppeteer from 'puppeteer';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));

(async () => {
  const browser = await puppeteer.launch({
    headless: 'new',
    args: ['--no-sandbox', '--disable-setuid-sandbox']
  });

  const page = await browser.newPage();
  await page.setViewport({ width: 1920, height: 1080 });

  // 截图 - 总览页面
  await page.goto('http://localhost:4173', { waitUntil: 'networkidle0' });
  await sleep(2000);
  await page.screenshot({
    path: path.join(__dirname, 'screenshot-overview.png'),
    fullPage: true
  });
  console.log('截图 1: 总览页面已保存');

  // 点击因子 Tab 截图
  await page.evaluate(() => {
    const buttons = document.querySelectorAll('button');
    buttons.forEach(btn => {
      if (btn.textContent.includes('因子')) btn.click();
    });
  });
  await sleep(1000);
  await page.screenshot({
    path: path.join(__dirname, 'screenshot-factors.png'),
    fullPage: true
  });
  console.log('截图 2: 因子页面已保存');

  // 点击回测 Tab 截图
  await page.evaluate(() => {
    const buttons = document.querySelectorAll('button');
    buttons.forEach(btn => {
      if (btn.textContent.includes('回测')) btn.click();
    });
  });
  await sleep(1000);
  await page.screenshot({
    path: path.join(__dirname, 'screenshot-backtest.png'),
    fullPage: true
  });
  console.log('截图 3: 回测页面已保存');

  await browser.close();
  console.log('所有截图已完成');
})();

File: web/screenshot.py
#!/usr/bin/env python3
"""
前端页面截图脚本
"""
from playwright.sync_api import sync_playwright
import time
import os

os.chdir(os.path.dirname(os.path.abspath(__file__)))

with sync_playwright() as p:
    print("启动浏览器...")
    browser = p.chromium.launch(headless=True)
    page = browser.new_page()
    page.set_viewport_size({"width": 1440, "height": 900})

    print("访问页面...")
    page.goto("http://localhost:4173", wait_until="networkidle", timeout=30000)

    print("等待页面渲染...")
    time.sleep(3)

    # 截图 1 - 总览页面
    print("截图 1: 总览页面")
    page.screenshot(path="screenshot-overview.png", full_page=False)

    # 截图 2 - 点击因子 Tab
    print("截图 2: 因子页面")
    page.click('button:has-text("因子")')
    time.sleep(2)
    page.screenshot(path="screenshot-factors.png", full_page=False)

    # 截图 3 - 点击回测 Tab
    print("截图 3: 回测页面")
    page.click('button:has-text("回测")')
    time.sleep(2)
    page.screenshot(path="screenshot-backtest.png", full_page=False)

    browser.close()
    print("截图完成！")
